CONTRIBUTING.md

```diff
@@ -0,0 +1,124 @@
+# Contributing to SkyPilot
+
+Thank you for your interest in contributing to SkyPilot! We welcome and value 
+all contributions to the project, including but not limited to: 
+
+* [Bug reports](https://github.com/skypilot-org/skypilot/issues) and [discussions](https://github.com/skypilot-org/skypilot/discussions)
+* [Pull requests](https://github.com/skypilot-org/skypilot/pulls) for bug fixes and new features
+* Test cases to make the codebase more robust
+* Examples
+* Documentation
+* Tutorials, blog posts and talks on SkyPilot
+
+Please read through this document before contributing so that there is all the necessary information needed for us to effectively respond to your contribution.
+
+## Contributing Code
+
+We use Github to track issues and features. For new contributors, we recommend looking at issues labeled ["good first issue"](https://github.com/sky-proj/sky/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22+).
+
+### Installing SkyPilot for development
+We recommend using editable mode (`-e`) when installing:
+```bash
+# SkyPilot requires python >= 3.6 and < 3.10.
+# You can just install the dependencies for
+# certain clouds, e.g., ".[aws,azure,gcp]"
+pip install -e ".[all]"
+pip install -r requirements-dev.txt
+```
+IMPORTANT: Please `export SKYPILOT_DEV=1` before running the CLI commands in the terminal, so that developers' usage logs do not pollute the actual user logs.
+
+### Environment variables for developers
+- `export SKYPILOT_DEV=1` to send usage logs to dev space.
+- `export SKYPILOT_DISABLE_USAGE_COLLECTION=1` to disable usage logging.
+- `export SKYPILOT_DEBUG=1` to show debugging logs (use logging.DEBUG level).
+- `export SKYPILOT_MINIMIZE_LOGGING=1` to minimize the logging for demo purpose.
+
+### Testing
+To run smoke tests:
+```
+bash tests/run_smoke_tests.sh
+
+# Run one of the smoke tests
+bash tests/run_smoke_tests.sh test_minimal
+```
+
+For profiling code, use:
```

Comment:
I think it's generally useful to have these snippets here and the contributing guide can be verbose in the interest of completeness. Also okay to remove if you feel strongly otherwise